### ISSUE TITLE
Can't even build: Incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.

### ISSUE THREAD

**Issue by marcchiu** [luanpotter/audioplayers#923]
Please help! I just created a new flutter project and have set the minimumsdkversion to 23 and pub added audioplayer: ^0.19.0. The app would not compile and keep giving this error, changing the ext.kotlin_version in android/build.gradle to '1.5.10' did not help:

`
Running Gradle task 'assembleDebug'...
e: Incompatible classes were found in dependencies. Remove them from the classpath or use '-Xskip-metadata-version-check' to suppress errors
e: /Users/marc/.gradle/caches/transforms-2/files-2.1/24fa3aa8d2270e5eb067bbe36e9b7563/jetified-kotlin-stdlib-1.5.10.jar!/META-INF/kotlin-stdlib.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.5.1, expected version is 1.1.15.

...VERY LONG ERRORS WITH THE SAME PROBLEM...

The class is loaded from /Users/marcus/.gradle/caches/transforms-2/files-2.1/24fa3aa8d2270e5eb067bbe36e9b7563/jetified-kotlin-stdlib-1.5.10.jar!/kotlin/Unit.class

FAILURE: Build failed with an exception.

 What went wrong:
Execution failed for task ':audioplayers:compileDebugKotlin'.
Compilation error. See log for more details
`

**Comment by iTomCat** [comment]
Hi in build.gradle(Project: android)
change from:
ext.kotlin_version = '1.3.50'
to:
ext.kotlin_version = '1.4.32'

**Comment by EastLee** [comment]
I have the same problem on mac! How to solve it ?

**Comment by iTomCat** [comment]
I had this problem on mac too. It doesn't matter because it's about the Kotlin version.
in the  program tree select android. Right mouse click and select Flutter/Open Android module in Android Studio.
then open Gradle Scripts/build.gradle(Project: android)
and in buildscript change  ext.kotlin_version:

buildscript {
from:
   ext.kotlin_version = '1.3.50'
to:
ext.kotlin_version = '1.4.32'
....

after thet click 'SyncNow', and back fo Flutter

**Comment by marcchiu** [comment]
It is working now! Thank you very much! May I ask why wasn't that ext.kotlin_version 1.5.10 but 1.4.32?

The link to the AudioCache documentation post in pub.dev is dead, should be this one https://github.com/luanpotter/audioplayers/blob/master/packages/audioplayers/doc/audio_cache.md

> I had this problem on mac too. It doesn't matter because it's about the Kotlin version.
> in the program tree select android. Right mouse click and select Flutter/Open Android module in Android Studio.
> then open Gradle Scripts/build.gradle(Project: android)
> and in buildscript change ext.kotlin_version:
> 
> buildscript {
> from:
> ext.kotlin_version = '1.3.50'
> to:
> ext.kotlin_version = '1.4.32'
> ....
> 
> after thet click 'SyncNow', and back fo Flutter

**Comment by Eazash** [comment]
Gradle sync is failing even after setting kotlin version to '1.4.32' with message 'Gradle sync failed: Sync failed: reason unknown'

> I had this problem on mac too. It doesn't matter because it's about the Kotlin version.
> in the program tree select android. Right mouse click and select Flutter/Open Android module in Android Studio.
> then open Gradle Scripts/build.gradle(Project: android)
> and in buildscript change ext.kotlin_version:
> 
> buildscript {
> from:
> ext.kotlin_version = '1.3.50'
> to:
> ext.kotlin_version = '1.4.32'
> ....
> 
> after thet click 'SyncNow', and back fo Flutter

**Comment by codetricity** [comment]
> It is working now! Thank you very much! May I ask why is that ext.kotlin_version 1.4.32 but not 1.5.10(or the most update version)?
> 
> The link to the AudioCache documentation post in pub.dev is dead, should be this one https://github.com/luanpotter/audioplayers/blob/master/packages/audioplayers/doc/audio_cache.md

I have AudioCache working now too.  Thank you.  Does [AudioPlayer](https://pub.dev/documentation/audioplayers/latest/audioplayers/AudioPlayer-class.html) also work with local sound files?  At the moment, I have all my sounds assets local and I am only using [AudioCache](https://pub.dev/documentation/audioplayers/latest/audioplayers/AudioCache-class.html).  If AudioPlayer works with local assets, should I be able to use [AudioPlayer.play](https://pub.dev/documentation/audioplayers/latest/audioplayers/AudioPlayer/play.html)?  There is a parameter for `isLocal`, which I set.

On Android 11, Pixel 3 emulator with local asset.

## Does not work

```dart
  TextButton(
      onPressed: () {
        audioPlayer.play(
            'C:\Users\craig\Documents\Development\practice\flutter\ap\assets\music.mp3',
            isLocal: true,
            volume: 1.0);
      },
      child: Text('play music')),
```

This is the error.

```
E/MediaPlayerNative(10594): error (1, -2147483648)
E/MediaPlayer(10594): Error (1,-2147483648)
E/MediaPlayerNative(10594): stop called in state 0, mPlayer(0xb0d58160)
E/MediaPlayerNative(10594): error (-38, 0)
V/MediaPlayer(10594): resetDrmState:  mDrmInfo=null mDrmProvisioningThread=null mPrepareDrmInProgress=false mActiveDrmScheme=false
V/MediaPlayer(10594): cleanDrmObj: mDrmObj=null mDrmSessionId=null
```

## works

```dart
  TextButton(
    child: Text('play  sound'),
    onPressed: () {
      audioCache.play('music.mp3', volume: .6);
    },
  ),
```

## system info

in `pubspec.yaml`

```yaml
environment:
  sdk: ">=2.12.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^1.0.2
  audioplayers: ^0.19.0


  assets:
    - assets/
    - assets/music.mp3
```

with flutter doctor

```
PS C:\Users\craig\Documents\Development\practice\flutter\ap> flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 2.2.2, on Microsoft Windows [Version 10.0.19043.1052], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.9.3)
[√] Android Studio (version 4.1.0)
[√] IntelliJ IDEA Community Edition (version 2021.1)
[√] VS Code (version 1.57.1)
[√] Connected device (4 available)
```

**Comment by sjrphx** [comment]
> In build.gradle(Project: android)
> change from:
> ext.kotlin_version = '1.3.50'
> to:
> ext.kotlin_version = '1.4.32'

Doesn't work for me as well. Please don't close this issue.

**Comment by luanpotter** [comment]
Fixed by #939

**Comment by TheJosh** [comment]
Maybe this could be added into the documentation or readme? The build version fix seems to be a common issue.
